setup.md

```diff
@@ -0,0 +1,72 @@
+# Setting up your local development environment for PyGMT
+
+We will use Git, Anaconda, and the `conda` package manager to install and
+manage the development environment.
+
+This guide assumes a basic understanding of git using the command line. For
+more help, attend the optional setup session on August 17 and/or review the
+[Software Carpentry](https://software-carpentry.org/)'s courses on
+[the Unix Shell](https://swcarpentry.github.io/shell-novice/) and
+[Version Control with Git](https://swcarpentry.github.io/git-novice/).
+
+## Cloning and forking the repository
```

Comment:
Is this the recommended workflow (clone, fork, and add a remote)?

When I contributed to open source projects that I don't have permission, I usually clone my own fork (e.g., seisman/pygmt) to my computer, and add the main repository (GenericMappingTools/pygmt) as a remote (e.g., `upstream`).

Reply:
I could change this to a fork then clone setup, if you think that's easier for contributors without write access. I chose this order because it is used in the reference recommended in gmt and pygmt's contributing guide (https://www.asmeurer.com/git-workflow/).

Reply:
OK. I think the current workflow is better, at least contributors don't have to worry about how to synchronize the upstream main branch to their own fork's main branch.